### PR TITLE
Make the `NotEnoughMoney` API error machine-readable.

### PR DESCRIPTION
**Pull request by jonathanknowles**
## Issue

ADP-3247

## Description

This PR revises the `NotEnoughMoney` API error so that it is machine-readable, with a well-defined schema.

## Example

```json
{ "code": "not_enough_money"
, "message": "I can't process this payment as there are not enough funds available in the wallet."
, "info":
  { "shortfall":
    { "ada":
      { "quantity": 1
      , "unit": "lovelace"
      }
    , "assets":
      [ { "policy_id": "FRUIT"
        , "asset_name": "APPLE"
        , "quantity": 2
        }
      , { "policy_id": "FRUIT"
        , "asset_name": "BANANA"
        , "quantity": 4
        }
      ]
    }
  }
}
```

### CODE REVIEW

File: specifications/api/swagger.yaml [hunk 1]
```diff
@@ -4460,10 +4460,25 @@ x-errNotEnoughMoney: &errNotEnoughMoney
   properties:
     message:
       type: string
-      description: May occur when there's not enough money in the wallet to cover a requested payment.
+      description: |
+        Indicates that there's not enough money in the wallet to cover a
+        requested payment.
```
Comment (jonathanknowles) [inline specifications/api/swagger.yaml]: Add a sentence that describes the `shortfall` field in more detail.
Emphasise that the shortfall just describes the assets that are missing, and that if supplied, would unblock whatever operation it was that the user was attempting.

Comment (jonathanknowles) [reply specifications/api/swagger.yaml]: Fixed in 0432db5cad7de78690601323617959a54ac12cea.

File: lib/wallet/integration/src/Test/Integration/Framework/TestData.hs [hunk 2]
```diff
@@ -378,11 +377,6 @@ errMsg403NotAnIcarusWallet =
     "I cannot derive new address for this wallet type.\
     \ Make sure to use a sequential wallet style, like Icarus."
 
-errMsg403NotEnoughMoney :: String
```
Comment (paweljakubas) [inline lib/wallet/integration/src/Test/Integration/Framework/TestData.hs:381]: :1st_place_medal:

File: lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs [hunk 3]
```diff
@@ -2472,10 +2472,10 @@ spec = describe "SHELLEY_TRANSACTIONS" $ do
         -- Try withdrawing when no not enough money
         rTx <- request @(ApiTransaction n) ctx
             (Link.createTransactionOld @'Shelley wSelf) Default payload
-        verify rTx
-            [ expectResponseCode HTTP.status403
-            , expectErrorMessage errMsg403NotEnoughMoney
-            ]
+        verify rTx [expectResponseCode HTTP.status403]
+        decodeErrorInfo rTx `shouldSatisfy` \case
```
Comment (paweljakubas) [inline lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs:2476]: :+1:

**Review by paweljakubas — APPROVED** [review]
very nice PR

**Pull request merged**